design/open/kv3-execution-plan.md

```diff
@@ -47,17 +73,21 @@ Moreover, as these are quite exhaustive operations, may be used as part of perfo
 * `branch_iterator`, sorted by branch name - trivial. sorted by commitID - TBD (still being discussed)
 * Repository Deletion - as suggested, maintaining all repository relevant entries under a unique partition will make it easy to delete as an entire partition. The repository key itself is outside of the partition (in a general common partition, to support listing) and will be deleted first. Once deleted, the entire partition will become unavailable, making the data state consistent either ways
 
+## Update Commits, staging and lock-free
+we will follow @arielshaqed suggestion to prevent commits overriding each other by using `KV`'s `SetIf` on the branch update, making sure no racing commit is already finished and being overridden. In case a commit fails to finish due to another commit "beating" it to the finish line - a retry should be attempted. This implementation will require to support multiple staging tokens, as proposed by [lakeFS on KV design proposal](https://github.com/treeverse/lakeFS/blob/b3204edad00f88f8eb98524ad940fde96e02ab0a/design/open/metadata_kv/index.md#committer-flow). Lock-free commits will be handled out of the scope of KV3 milestone
```

Comment:
Actually Ariel's suggestion was the `Lock-free commits`

Reply:
Eventually figured it out. By Ariel's suggestion, I was referring to [this comment](https://github.com/treeverse/lakeFS/pull/3566#discussion_r907073940) and you were referring to #3091
Modified the doc to clarify that